v3/newrelic/collector.go

```diff
@@ -136,11 +137,10 @@ func rpmURL(cmd rpmCmd, cs rpmControls) string {
 	return u.String()
 }
 
-func compress(b []byte) (*bytes.Buffer, error) {
+func compress(b []byte, w *gzip.Writer) (*bytes.Buffer, error) {
```

Comment:
If you access to a pointer gzip writer concurrently, it will probably cause race condition.